### PR TITLE
Sa/dockerize

### PR DESCRIPTION
**Pull request by stevejuma**
* Dockerizing the availability app. 
* Switching config to a separate class to get away for HoconConfig Hell
* Updated Gradle version 
* Updated Gradle properties to be inline with the build.gradle file 
* Some minor refactoring

### CODE REVIEW

File: src/main/kotlin/com/alphasights/availability/config/AppConfig.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package com.alphasights.availability.config
+
+import java.util.UUID
+import mu.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
+
+data class AppConfig(
+    val app: App,
+    val dynamo: DynamoDbProperties
```
[the remark this answers is not in the document]
Comment (stevejuma) [reply src/main/kotlin/com/alphasights/availability/config/AppConfig.kt:10]: I think it's okay to namespace the config properties into their own name. I image if we implement a postgres one we would have `postgres` as the database key.

File: src/test/resources/application.conf [hunk 2]
```diff
@@ -6,22 +6,4 @@ ktor {
   application {
```
[the remark this answers is not in the document]
Comment (stevejuma) [reply src/test/resources/application.conf:6]: `ktor` still requires some values provided like the port and module definition so don't think we can get rid of this just yet. It's now just a static file that isn't referenced elsewhere.

File: src/main/kotlin/com/alphasights/availability/Application.kt [hunk 3]
```diff
@@ -48,21 +48,42 @@ fun Application.module(testing: Boolean = false) {
...
+                    status = HttpStatusCode.NotFound.value
+                )
+            )
         }
+
         exception<InvalidAdvisorId> { _ ->
-            call.respond(HttpStatusCode.BadRequest)
+            call.respond(
+                HttpStatusCode.BadRequest, ApiError(
+                    message = "Invalid advisor id",
```
[the remark this answers is not in the document]
Comment (stevejuma) [reply src/main/kotlin/com/alphasights/availability/Application.kt:75]: Good point. We could get rid of the `InvalidAdvisorId` class and just throw an ApiErrorException with the same message. That way removing this piece of code here.

File: src/test/resources/application.conf (hunk 2 above)
[the remark this answers is not in the document]
Comment (PrasannaGnanaraj) [reply src/test/resources/application.conf:6]: ah okay

File: src/main/kotlin/com/alphasights/availability/Application.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (PrasannaGnanaraj) [reply src/main/kotlin/com/alphasights/availability/Application.kt:75]: 👍

File: src/main/kotlin/com/alphasights/availability/config/AppConfig.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (PrasannaGnanaraj) [reply src/main/kotlin/com/alphasights/availability/config/AppConfig.kt:10]: hmm alrighty

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+dynamo:
+  accessKey: ${DYNAMO_ACCESS_KEY}
+  secretKey: ${DYNAMO_SECRET_KEY}
+  endPoint: ${DYNAMO_ENDPOINT}
+  region: ${DYNAMO_REGION}
+  tables:
+    - className: com.alphasights.availability.db.models.Availability
+      readCapacity: 1
```
[the remark this answers is not in the document]
Comment (stevejuma) [reply src/main/resources/application.yaml]: This has been updated now.

**Pull request merged**